code/modules/spell_system/spells/spells.dm

```diff
@@ -34,8 +34,7 @@
 			H.add_spell(spell_to_add)
 
 /mob/proc/add_spell(var/spell/spell_to_add, var/spell_base = "wiz_spell_ready", var/master_type = /obj/screen/movable/spell_master)
-	if(isobserver(usr)) // we spawned in via ghost spawner, so set our thing to this so we don't screw up
-		usr = src
+	usr = src // whoever made screen objects is an asshole and forced me to do this, i regret absolutely nothing
```

Comment:
what's this for, again?

Reply:
this lets the spell master code figure out where exactly to put the spell's icon in the spell hotbar. without it, it's placed in the far bottom left and stretches the screen until you close and reopen the spell list. it assumes usr is passed, because you always clicked on the wizard spellbook to get the spell, which passed usr